crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs

```diff
@@ -0,0 +1,235 @@
+    let rollup_cell = CellOutput::new_builder()
+        .type_(Some(rollup_type_script.clone()).pack())
+        .build();
+    let mut chain = setup_chain(rollup_type_script.clone()).await;
+
+    // Deposit accounts
+    let accounts: Vec<_> = (0..ACCOUNTS_COUNT)
+        .map(|_| random_always_success_script(&rollup_script_hash))
+        .collect();
+    let deposits = accounts.iter().map(|account_script| {
+        DepositRequest::new_builder()
+            .capacity(DEPOSIT_CAPACITY.pack())
+            .sudt_script_hash(H256::zero().pack())
+            .amount(0.pack())
+            .script(account_script.to_owned())
+            .registry_id(gw_common::builtins::ETH_REGISTRY_ACCOUNT_ID.pack())
+            .build()
+    });
+
+    let block_result = {
+        let mem_pool = chain.mem_pool().as_ref().unwrap();
+        let mut mem_pool = mem_pool.lock().await;
+        construct_block(&chain, &mut mem_pool, deposits.clone().collect())
+            .await
+            .unwrap()
+    };
+    let apply_deposits = L1Action {
+        context: L1ActionContext::SubmitBlock {
+            l2block: block_result.block.clone(),
+            deposit_requests: deposits.collect(),
+            deposit_asset_scripts: Default::default(),
+            withdrawals: Default::default(),
+        },
+        transaction: build_sync_tx(rollup_cell.clone(), block_result),
+        l2block_committed_info: L2BlockCommittedInfo::new_builder()
+            .number(1u64.pack())
+            .build(),
+    };
+    let param = SyncParam {
+        updates: vec![apply_deposits],
+        reverts: Default::default(),
+    };
+    chain.sync(param).await.unwrap();
+    assert!(chain.last_sync_event().is_success());
+
+    // Generate withdrawals
+    let mut withdrawals: Vec<_> = {
+        accounts
+            .iter()
+            .map(|account_script| {
+                let owner_lock = Script::default();
+                let raw = RawWithdrawalRequest::new_builder()
+                    .capacity(WITHDRAWAL_CAPACITY.pack())
+                    .account_script_hash(account_script.hash().pack())
+                    .sudt_script_hash(H256::zero().pack())
+                    .owner_lock_hash(owner_lock.hash().pack())
+                    .registry_id(gw_common::builtins::ETH_REGISTRY_ACCOUNT_ID.pack())
+                    .build();
+                let withdrawal = WithdrawalRequest::new_builder().raw(raw).build();
+                WithdrawalRequestExtra::new_builder()
+                    .request(withdrawal)
+                    .owner_lock(owner_lock)
+                    .build()
+            })
+            .collect()
+    };
+    let pending_withdrawals = withdrawals.split_off(withdrawals.len() - 5);
+    let mem_block_withdrawals = withdrawals;
+
+    // Insert error nonce withdrawal and expect them to be removed during pending restore
+    let invalid_withdrawals_count = pending_withdrawals.len() - 5;
```

Comment:
`invalid_withdrawals_count` here would be zero?

Reply:
This count is used to create invalid withdrawals for test, should not be zero.